plugins/woocommerce-admin/client/products/product-page.tsx

```diff
@@ -1,19 +1,73 @@
 /**
  * External dependencies
  */
-import { __experimentalEditor as Editor } from '@woocommerce/product-editor';
+import {
+	__experimentalEditor as Editor,
+	AUTO_DRAFT_NAME,
+} from '@woocommerce/product-editor';
 import { Product } from '@woocommerce/data';
+import { useDispatch, useSelect, select as WPSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { Spinner } from '@wordpress/components';
+import { useParams } from 'react-router-dom';
 
 /**
  * Internal dependencies
  */
 import './product-page.scss';
 
-const dummyProduct = {
-	name: 'Example product',
-	short_description: 'Short product description content',
-} as Product;
+const ProductEditor: React.FC< { product: Product | undefined } > = ( {
+	product,
+} ) => {
+	if ( ! product ) {
+		return <Spinner />;
+	}
+
+	return <Editor product={ product } settings={ {} } />;
+};
+
+const EditProductEditor: React.FC< { productId: string } > = ( {
+	productId,
+} ) => {
+	const { product } = useSelect( ( select: typeof WPSelect ) => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore Missing types.
+		const { getEditedEntityRecord } = select( 'core' );
+
+		return {
+			product: getEditedEntityRecord( 'postType', 'product', productId ),
+		};
+	} );
+
+	return <ProductEditor product={ product } />;
+};
+
+const AddProductEditor = () => {
+	const { saveEntityRecord } = useDispatch( 'core' );
+	const [ product, setProduct ] = useState< Product | undefined >(
+		undefined
+	);
+
+	useEffect( () => {
+		saveEntityRecord( 'postType', 'product', {
+			title: AUTO_DRAFT_NAME,
+			status: 'auto-draft',
```

Comment:
Setting the status to `auto-draft` will prevent it from being displayed in listings, and should auto-delete it after 7 days (haven't tested that bit yet).